src/Data/Project.hs

```diff
@@ -43,18 +41,6 @@ newtype ProjectException
   = FileNotFound FilePath
     deriving (Show, Eq, Typeable, Exception)
 
-readFile :: (Member (Error SomeException) sig, Carrier sig m)
-         => Project
-         -> File
-         -> m (Maybe Blob)
-readFile Project{..} f =
-  let p         = filePath f
-      candidate = find (\b -> blobPath b == p) projectBlobs
-  in if
-    | p == "/dev/null"  -> pure Nothing
```

Comment:
We don’t appear to be checking for this in the replacement code; is that ok? When would we have received `/dev/null` files as inputs here?

Reply:
The “/dev/null” path is only for diffs, right? Diffing code never calls this function, so I think that it’s fine (n.b. `readBlobFromFile` handles the case of `/dev/null` elsewhere).